frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts

```diff
@@ -290,3 +310,18 @@ export const useChartEvents = (
     eventHandlers,
   };
 };
+
+function getTransformedDatumIndex(
+  transformedDataset: ChartDataset,
+  originalDatumIndex: number,
+) {
+  const transformedDatumIndex = transformedDataset.findIndex(
```

Comment:
Potentially we may want to change it in future if we notice any hover performance issues on large datasets

Reply:
Agreed. Maybe we could update the original `dataset` points with their corresponding `transformedDataset` index, so that finding the correct `transformedDataset` index is just a lookup on that original data point.

i.e. Switching up which "dataset" contains the others mapped index. Instead of `transformedDataset` containing the index mapped to the original, the original would contain the index mapped to its corresponding `transformedDataset` index.